RecoBTag/DeepFlavour/interface/sorting_modules.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * sorting_modules.h
+ *
+ *  Created on: 5 Mar 2017
+ *      Author: jkiesele
+ */
+
+#ifndef DEEPNTUPLES_DEEPNTUPLIZER_INTERFACE_SORTING_MODULES_H_
+#define DEEPNTUPLES_DEEPNTUPLIZER_INTERFACE_SORTING_MODULES_H_
+
+#include <algorithm>
+#include <cmath>
+#include <vector>
+
+namespace sorting{
+
+/*
+ * the std::sort function needs a strict ordering.
+ * means if your_function(a,b) is true, than your_function(b,a) must be false for all a and b
+ */
+
+template<class T>
+bool comparePt(T a, T b){
+    if(!a)return true;
+    if(!b)return false;
+    return a->pt()<b->pt();
+}
+
+
+template <class T>
+class sortingClass{
+public:
+
+    sortingClass():sortValA(0),sortValB(0),sortValC(0),t_(0){}
+
+    sortingClass(const T& t, float sortA, float sortB=0, float sortC=0){
+        t_=t;
+        sortValA=sortA;
+        sortValB=sortB;
+        sortValC=sortC;
+    }
+    sortingClass(const sortingClass&rhs):
+    sortValA(rhs.sortValA),sortValB(rhs.sortValB),sortValC(rhs.sortValC),t_(rhs.t_)
+    {	}
+    
+    sortingClass& operator=(const sortingClass&rhs){
```

Comment:
is there a need for an explicit copy operator and copy constructor (different from what the compiler makes)?

Reply:
checked with @jkieseler and not needed anymore, will remove

Reply:
removed with https://github.com/cms-sw/cmssw/pull/19893/commits/6ad48f2a3527cf49f3a753cf57bfe30380048472